lib/discourse_translator/locale_normalizer.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module DiscourseTranslator
+  class LocaleNormalizer
+    # Normalizes locale string, matching the list of I18n.locales where possible
+    # @param locale [String,Symbol] the locale to normalize
+    # @return [String] the normalized locale
+    def self.normalize_to_i18n(locale)
+      return nil if locale.blank?
+      locale = locale.to_s.gsub("-", "_")
+
+      i18n_pairs.each { |downcased, value| return value if locale.downcase == downcased }
```

Comment:
```suggestion
      i18n_pairs.find { |downcased, value| :en == downcased }.last
```

Technically same, but `find` gives me clear indicator we are searching for one value instead of looping through.

Please feel free to ignore as I feel it is just preference :)